apps/desktop/src-tauri/src/inference/server.rs

```diff
@@ -58,24 +58,55 @@ async fn post_completions(payload: Json<CompletionRequest>) -> impl Responder {
 
   let (token_sender, receiver) = flume::unbounded::<Bytes>();
 
-  HttpResponse::Ok()
-    .append_header(("Content-Type", "text/event-stream"))
-    .append_header(("Cache-Control", "no-cache"))
-    .keep_alive()
-    .streaming({
-      let abort_flag = Arc::new(RwLock::new(false));
-
-      AbortStream::new(
-        receiver,
-        abort_flag.clone(),
-        start(InferenceThreadRequest {
-          model_guard: model_guard.clone(),
-          abort_flag: abort_flag.clone(),
-          token_sender,
-          completion_request: payload.0,
-        }),
-      )
-    })
+  if let Some(true) = payload.stream {
```

Comment:
This should be payload.0.stream I think, since it's a JSON.

If we can reconcile our trait above, we ca infer the stream boolean via the `completion_request` as well, skipping a couple of lookup hoop!